### PR TITLE
Introduce a separate symbol kind for enum members

### PR DESCRIPTION
**Pull request by pubudu91**
## Purpose
This PR introduces a new symbol kind for distinguishing between regular constants and constants which are members of enums.
 
## Approach
- Introduce a separate symbol kind named `ENUM_MEMBER`
- For `ConstantSymbol` instances who happen to be enum members, `kind()` returns `ENUM_MEMBER` instead of `CONSTANT`.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by IMS94 — DISMISSED** [review]
LGTM!

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ public void sort(BallerinaCompletionContext context, T node, List<LSCompletionIt
             }
             if (symbol.kind() == FUNCTION || symbol.kind() == METHOD) {
                 completionItems.add(populateBallerinaFunctionCompletionItem(symbol, ctx));
-            } else if (symbol.kind() == SymbolKind.CONSTANT) {
+            } else if (symbol.kind() == SymbolKind.CONSTANT || symbol.kind() == SymbolKind.ENUM_MEMBER) {
                 CompletionItem constantCItem = ConstantCompletionItemBuilder.build((ConstantSymbol) symbol, ctx);
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java:161]: For the enum member completion items shall we set the completion item kind as `EnumMember`

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

**Review by mohanvive — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java (hunk 1 above)
Comment (pubudu91) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java:161]: +1

Comment (pubudu91) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java:161]: Fixed with 00773b8

**Review by mohanvive — APPROVED** [review]

**Review by nadeeshaan — APPROVED** [review]

**Pull request merged**